static/js/import_wizard2/templates.tsx

```diff
@@ -238,6 +238,43 @@ function getConstantVarUserMapping(predictedMapping: Mapping): Mapping {
   return userMapping;
 }
 
+function getSingleVarMultiDateUserMapping(predictedMapping: Mapping): Mapping {
+  let userMapping = new Map();
+  if (!_.isEmpty(predictedMapping)) {
+    userMapping = _.clone(predictedMapping);
+    const nonColumnMappedThings = new Set([
+      MappedThing.STAT_VAR,
+      MappedThing.DATE,
+    ]);
+    predictedMapping.forEach((mappingVal, mappedThing) => {
+      let isInvalid = false;
+      // mappingVal for stat var must have fileConstant
+      if (
+        mappedThing === MappedThing.STAT_VAR &&
+        _.isEmpty(mappingVal.fileConstant)
+      ) {
+        isInvalid = true;
+      }
+      // mapping val for date must have headers
+      isInvalid =
+        isInvalid ||
+        (mappedThing === MappedThing.DATE && _.isEmpty(mappingVal.headers));
+      // mappingVal for all other mapped things must have a column
+      isInvalid =
+        isInvalid ||
+        (!nonColumnMappedThings.has(mappedThing) &&
+          _.isEmpty(mappingVal.column));
+      if (isInvalid) {
+        console.log(
+          `Invalid mappingVal for ${mappedThing}. Entry deleted from mapping.`
+        );
+        userMapping.delete(mappedThing);
```

Comment:
Is it worth propagating this to UI?  Maybe have a TODO

Reply:
I don't think we should show this on the UI because this is validation on predictions that we have generated from the user file. This has nothing to do with the user's input, there's nothing for the user to do, and user shouldn't need to have insight on the predictions we make about their file.

Reply:
Makes sense.